pkg/planner/core/plan_cache.go

```diff
@@ -568,13 +579,23 @@ func rebuildRange(p Plan) error {
 				if err != nil {
 					return err
 				}
-				x.Handles[i] = kv.IntHandle(iv)
+				if i >= len(x.Handles) {
+					// Previous execution might have removed duplicates,
+					// so Handles are not the same length as HandleParams
+					x.Handles = append(x.Handles, kv.IntHandle(iv))
```

Comment:
Better to return an error directly for safety.

Reply:
Done by checking it before the loop.

Reply:
I added a check and return an error in case len(x.Handles) != len(x.HandleParams)